src/interactive-window/interactiveWindow.ts

```diff
@@ -618,25 +553,31 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         isDebug: boolean
     ) {
         traceInfoIfCI('InteractiveWindow.ts.createExecutionPromise.start');
-        // Wait for the cell to get created (required to insert error messages into cell output).
-        // When we wait for kernel to start, its possible it will fail, and
-        // higher up, we insert errors into the cell (but the cell needs to have been inserted).
+        // Kick of starting kernels early.
+        const kernelPromise = this.startKernel();
```

Comment:
This is what fixes #9444, when we run each cell we need to trigger the kernel start.
That's what prompted most of this refactor